test/tests/functional/pbs_mom_dynamic_resource.py

```diff
@@ -338,8 +349,124 @@ def test_alter_res_value(self):
+        self.scheduler.set_sched_config({'mom_resources': 'foo'},
+                                        validate=True)
+        self.scheduler.add_resource('foo')
+        self.mom.add_config({'foo': mom_config_str})
+
+        # give write permission to group and others
+        self.du.chmod(path=fp, mode=0766)
+        self.check_access_log(fp)
+
+        # give write permission to group
+        self.du.chmod(path=fp, mode=0764)
+        self.check_access_log(fp)
+
+        # give write permission to others
+        self.du.chmod(path=fp, mode=0746)
+        self.check_access_log(fp)
+
+        # give write permission to user only
+        self.du.chmod(path=fp, mode=0744)
+        self.check_access_log(fp, exist=False)
+
+        # Create a script in home directory which has more open privileges
+        # This should make loading of this file fail in all cases
+        dir_temp = self.du.mkdtemp(mode=0766, dir=home_dir)
+        fp = self.du.create_temp_file(body=scr_body, dirname=dir_temp)
+        # Add to filenames for cleanup
+        self.filenames.append(fp)
+        self.filenames.append(dir_temp)
+
+        mom_config_str = '!' + fp
+        self.mom.add_config({'foo': mom_config_str})
+
+        # give write permission to group and others
+        self.du.chmod(path=fp, mode=0766)
+        self.check_access_log(fp)
+
+        # give write permission to group
+        self.du.chmod(path=fp, mode=0764)
+        self.check_access_log(fp)
+
+        # give write permission to others
+        self.du.chmod(path=fp, mode=0746)
+        self.check_access_log(fp)
+
+        # give write permission to user only
+        self.du.chmod(path=fp, mode=0744)
+        self.check_access_log(fp)
+
+        attr = {"type": "long", "flag": "h"}
+        self.server.manager(MGR_CMD_CREATE, RSC, attr,
+                            id="foo", expect=True)
+        # Create a dynamic script with right permissions
+        fp = self.du.create_temp_file(body=scr_body, dirname=home_dir)
+        # Add to filenames for cleanup
+        self.filenames.append(fp)
+
+        mom_config_str = '!' + fp
+        self.mom.add_config({'foo': mom_config_str})
+        # give write permission to user only
+        self.du.chmod(path=fp, mode=0744)
+        self.check_access_log(fp, exist=False)
+
+        match_from = time.time()
+        time.sleep(1)
+        # give write permission to others
+        self.du.chmod(path=fp, mode=0746)
+        # submit a job requesting this mom resource
+        attr = {"Resource_List.foo": 4}
+        j = Job(TEST_USER, attrs=attr)
+        self.server.submit(j)
+        self.mom.log_match(fp + ' file has a non-secure file access',
+                           starttime=match_from, existence=True,
+                           max_attempts=10)
+
+        # Create dynamic resource script in tmp directory and check
+        # file permissions
+        fp = self.du.create_temp_file(body=scr_body, dirname=home_dir)
```

Comment:
You are creating the script in home directory and not tmp, right? How is this section of checks different from the checks that you've already performed earlier in this test?

Reply:
Ahh... that dirname should not be there. I'll remove it